File: lab_2/Task_1/isb_sequence_gen.cpp
#include <fstream>
#include <iostream>
#include <random>
#include <string>


/**
 * @brief Generates a random 128-bit sequence represented as a binary string.
 *
 * The function uses a Mersenne Twister pseudo-random number generator (std::mt19937)
 * with a hardware-based seed from std::random_device. Produces a string of 128 '0' and '1' characters.
 *
 * @return std::string - 128-character string consisting of '0's and '1's
 */
std::string generate_128_bit_sequence() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<size_t> number(0, 1);

    std::string result;
    for (size_t i = 0; i < 128; ++i) {
        result += std::to_string(number(gen));
    }
    return result;
}

/**
 * @brief Saves a string to a specified file path.
 *
 * @param dir std::string - Path where the file will be created/overwritten
 * @param string_to_save std::string - Content to write to the file
 * @throws std::ios_base::failure If file operations fail (exceptions enabled on output stream)
 */
void save_string(std::string dir, std::string string_to_save) {
    std::ofstream out;
    out.open(dir);
    out << string_to_save << std::endl;
    out.close();
}

/**
 * @brief Main program that generates a 128-bit sequence and saves it to a file
 *
 * Generates a binary sequence using generate_128_bit_sequence(), saves it to "cpp_sequence.txt",
 * and prints the sequence to standard output.
 */
int main() {
    std::string seq = generate_128_bit_sequence();
    save_string("cpp_sequence.txt", seq);
    std::cout << seq;
}

File: lab_2/Task_1/isb_sequence_gen.java
import java.security.SecureRandom;


/**
 * Class for generating cryptographically secure 128-bit sequences
 * and outputting them as binary-formatted strings.
 */
public class Main {

    /**
     * Main method that generates and prints a 128-bit binary sequence
     *
     * @param args Command-line arguments (not used in this implementation)
     *
     * @implSpec
     * Uses SecureRandom for cryptographically strong random number generation.
     * Generates 128 bits represented as '0'/'1' characters in a StringBuilder.
     * Outputs the resulting binary string to standard output.
     */
    public static void main(String[] args) {
        final SecureRandom random = new SecureRandom();
        final StringBuilder binaryString = new StringBuilder(128);

        // Generate 128 cryptographically secure random bits
        for (int i = 0; i < 128; i++) {
            final boolean bit = random.nextBoolean();
            binaryString.append(bit ? '1' : '0');
        }

        System.out.println(binaryString.toString());
    }
}